modules/Uri_Checker.py

```diff
@@ -10,43 +10,43 @@
 #https://github.com/theInfectedDrake/XSRFProbe
 
 
-import urllib.parse
 import re
-from core.colors import * # import ends
+from urllib.parse import urlsplit
+from core.colors import O # import ends
 
 def buildUrl(url, href): # receive form input type / url
 
-    if re.search('logout',href) or re.search('action=out',href) or re.search('action=logoff', href) or re.search('action=delete',href) or re.search('UserLogout',href) or re.search('osCsid', href) or re.search('file_manager.php',href) or href=="http://localhost": # make exclusion list
+    exclusions = 'logout action=out action=logoff action=delete UserLogout osCsid file_manager.php'
+    if href == "http://localhost" or any(s in href for s in exclusions.split()):
```

Comment:
```suggestion
    if href == "http://localhost" or any((re.search(s,href)) for s in exclusions.split()):
```

Reply:
You might consider this :D

Reply:
What values of sub and what values of s will break the last two lines?
```python
sub = '#'
for s in ('#', '\n#', '#\n', '\n#\n'):
    assert bool(re.match('#', s)) == s.startswith('#')
    assert bool(re.search('#', s)) == ('#' in s)
```

Reply:
Yes...  Only when you have a regex...  When you have normal strings (all str are Unicode in Py3) then the assert statements pass.